### PR TITLE
Django 4.0 and Python 3.10 support

### LINKED ISSUE #462: support for python 3.9 and 3.10
Add support for python versions 3.9 and 3.10 ?

### LINKED ISSUE #465: Add support for Django 4.0
First, Thanks for this great project! 👍 

### Summary:

Current django-celery-beat requires Django (>=2.2,<4.0) and hence doesn't install together w/ Django 4.0.

### LINKED ISSUE #518: zoneinfo.ZoneInfo' object has no attribute 'normalize'
### Summary:

When starting a celery beat container with latest master the following is seen in our logs. Are we missing some dependency?

[celery-beat-container] [2022-02-27 19:21:52,007: INFO/MainProcess] beat: Starting...
[celery-beat-container] [2022-02-27 19:21:52,045: CRITICAL/MainProcess] beat raised exception <class 'AttributeError'>: AttributeError("'zoneinfo.ZoneInfo' object has no attribute 'normalize'")
[celery-beat-container] Traceback (most recent call last):
[celery-beat-container]   File "/usr/local/lib/python3.10/site-packages/celery/apps/beat.py", line 105, in start_scheduler
[celery-beat-container]     service.start()
[celery-beat-container]   File "/usr/local/lib/python3.10/site-packages/celery/beat.py", line 645, in start
[celery-beat-container]     interval = self.scheduler.tick()
[celery-beat-container]   File "/usr/local/lib/python3.10/site-packages/celery/beat.py", line 343, in tick
[celery-beat-container]     self.populate_heap()
[celery-beat-container]   File "/usr/local/lib/python3.10/site-packages/celery/beat.py", line 317, in populate_heap
[celery-beat-container]     is_due, next_call_delay = entry.is_due()
[celery-beat-container]   File "/usr/local/lib/python3.10/site-packages/django_celery_beat/schedulers.py", line 136, in is_due
[celery-beat-container]     return self.schedule.is_due(last_run_at_in_tz)
[celery-beat-container]   File "/usr/local/lib/python3.10/site-packages/django_celery_beat/tzcrontab.py", line 45, in is_due
[celery-beat-container]     rem_delta = self.remaining_estimate(last_run_at)
[celery-beat-container]   File "/usr/local/lib/python3.10/site-packages/celery/schedules.py", line 610, in remaining_estimate
[celery-beat-container]     return remaining(*self.remaining_delta(last_run_at, ffwd=ffwd))
[celery-beat-container]   File "/usr/local/lib/python3.10/site-packages/celery/schedules.py", line 548, in remaining_delta
[celery-beat-container]     now = self.maybe_make_aware(self.now())
[celery-beat-container]   File "/usr/local/lib/python3.10/site-packages/celery/schedules.py", line 67, in now
[celery-beat-container]     return (self.nowfun or self.app.now)()
[celery-beat-container]   File "/usr/local/lib/python3.10/site-packages/django_celery_beat/tzcrontab.py", line 31, in nowfunc
[celery-beat-container]     return self.tz.normalize(
[celery-beat-container] AttributeError: 'zoneinfo.ZoneInfo' object has no attribute 'normalize'

* Celery-Beat Version: master branch as of 2022-02-27

### PR DESCRIPTION
**Pull request by ElSaico**
Closes #462 (needed to upgrade pytest because of pytest-dev/pytest#8546)
Closes #465 (removed all pytz-isms and fixed incompatibilities with django-timezone-field 5.0)
Closes #518 (was one of such pytz-isms)

### CODE REVIEW

File: t/proj/urls.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
-from django.conf.urls import url
+from django.urls import path
 from django.contrib import admin
 
 urlpatterns = [
-    url(r'^admin/', admin.site.urls),
+    path(r'admin/', admin.site.urls),
```
Comment (auvipy) [inline t/proj/urls.py:5]:
```suggestion
    path('admin/', admin.site.urls),
```

File: requirements/default.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 celery>=5.2.3,<6.0
-django-timezone-field>=4.2.3
+django-timezone-field>=5.0
+tzdata
 python-crontab>=2.3.4
```
Comment (auvipy) [inline requirements/default.txt:4]: backports.zoneinfo;python_version<"3.9" should this be added as well?

**Review by auvipy — CHANGES_REQUESTED** [review]

File: requirements/default.txt (hunk 2 above)
Comment (ElSaico) [reply requirements/default.txt:4]: This is added by django-timezone-field already, though I agree it wouldn't hurt to make explicit.

(especially as we do import it ourselves)

**Comment by hassanselim0** [comment]
> my only concern is does django below 4 support zoneinfo?

According to this: https://docs.djangoproject.com/en/4.0/releases/4.0/#zoneinfo-default-timezone-implementation
Django 3.2 was the first version to support non-`pytz` time zones, which I assume means that it would support `zoneinfo`.
This means you'll need to update the dependency from `>=2.2` to `>=3.2`.

File: .travis.yml [hunk 3]
```diff
@@ -16,11 +16,11 @@ python:
   - 3.7
   - 3.8
   - 3.9
+  - 3.10
 env:
   - DJANGO=2.2
-  - DJANGO=3.0
```
Comment (auvipy) [inline .travis.yml:21]: our travis CI don't work anymore, I am working on moving to github actions

**Comment by nickmccullum** [comment]
Any update on this? This is blocking us from upgrading one of our projects to Django 4, and we would love to get it merged.

**Review by auvipy — CHANGES_REQUESTED** [review]
needs rebase

**Comment by auvipy** [comment]
builds are failing

**Comment by ElSaico** [comment]
The build errors seem to be related to RabbitMQ being down during tests - except for the 3.8 one, which seems to call the 3.9 executable (and obviously fail) for whatever reason.

**Comment by tremechus** [comment]
@ElSaico checking in, looks like there is a rebase request in order to move this forward, is that something you can push?

**Comment by ElSaico** [comment]
@auvipy [Made a number of fixes in GitHub Actions](https://github.com/ElSaico/django-celery-beat/actions/runs/2435218458), including stuff that breaks on master already.

The only things broken now are the Sphinx builds, because of this:
```
Warning, treated as error:
  extlinks: Sphinx-6.0 will require a caption string to contain exactly one '%s' and all other '%' need to be escaped as '%%'.
```

I've fixed the entries on `conf.py`, but it seems like `sphinx_celery` has the same problem anyway. Thus we have two choices:

- Drop `-W` from the `sphinx-build` calls, avoiding erroring out on warnings
- Fix the `extlinks` entries in `sphinx_celery`

**Comment by auvipy** [comment]
> * Fix the `extlinks` entries in `sphinx_celery`

Fix the extlinks entries in sphinx_celery -- should be the go :)

**Comment by auvipy** [comment]
python 3.8 builds seems to be a github network issue

**Review by auvipy — APPROVED** [review]

**Comment by auvipy** [comment]
we can fix the sphinx issues in another PR

**Pull request merged**